appcelerator/titanium_mobile_windows
# [TIMOB-24611] Support for <SDKReference> and <PackageDependency>

[TIMOB-24611](https://jira.appcelerator.org/browse/TIMOB-24611)

Add ability to specify `<SDKReference>` and `<PackageDependency>`.

~~Please refer to commit https://github.com/appcelerator/titanium_mobile_windows/commit/c8112ae77c240ff903fc27853c4279c1384c181d for code review because this PR includes update for CMake 3.8 (that has thousand of files updated).~~

**tiapp.xml**
```xml
  <windows>
    <manifest>
      <SDKReferences>
       <SDKReference Include="Microsoft.VCLibs.120, Version=14.0" />
      </SDKReferences>
      <Dependencies>
        <PackageDependency Name="Microsoft.VCLibs.120.00.Debug.Universal" MinVersion="12.0.30501.0" Publisher="CN=Microsoft Corporation, O=Microsoft Corporation, L=Redmond, S=Washington, C=US" />
      </Dependencies>
    </manifest>
  </windows>
```

**How to test**

1) Build an app with `--build-only`

```
appc run -p windows --wp-sdk 10.0 --target ws-local -l trace --build-only
```

2) Go to `C:\Users\USERNAME\.titanium\vsbuild\YOUR_PROJECTNAME`

3) Following entry should be there in `Package.win10.appxmanifest.in`

```xml
  <Dependencies>
    <TargetDeviceFamily Name="Windows.Universal" MinVersion="10.0.0.0" MaxVersionTested="10.0.0.0" />
    <PackageDependency Name="Microsoft.VCLibs.120.00.Debug.Universal" MinVersion="12.0.30501.0" Publisher="CN=Microsoft Corporation, O=Microsoft Corporation, L=Redmond, S=Washington, C=US"/>
  </Dependencies>
```

4) Go to `C:\Users\USERNAME\.titanium\vsbuild\YOUR_PROJECTNAME\win10.x86`

5) Following entry should be there in `YOUR_PROJECTNAME.vcxproj`

```xml
  <ItemGroup>
    <SDKReference Include="Microsoft.VCLibs.120, Version=14.0"/>
  </ItemGroup>
```
